add media query debugger and css modules

phritolay opened the pull request:
Moving from Docz to Storybook. ~Move sass and postcss configurations here for reuse.~ Doing this ~also~ resolves conflicts between the theme and the project that occurs with CSS modules.

resolves #3
resolves #13

This PR is related to ample/gatsby-starter-ample#15

phritolay committed:
add media query debugger and css modules

Move sass and postcss configurations here for reuse. Doing this also resolves conflicts between the theme and the project that occurs with CSS modules.

resolves #3
resolves #13

This PR is related to ample/gatsby-starter-ample#15

phritolay committed:
add config and script for stylelint

resolves #13

phritolay committed:
move from docz to storybook, to resolve confilcts

bewteen the theme and the project

seancdavis commented on index.js:
```diff
@@ -1,3 +1,5 @@
 import Link from "./src/components/link"
+import DebugMediaQueries from "./src/components/layout/debug/media-queries"
 
 export { Link }
+export { DebugMediaQueries }
```
You can put these into one line:

```js
export { Link, DebugMediaQueries }
```

Also, make sure you're updated against master. I added SEO component and I think you'll run into an issue here.

seancdavis commented on package.json:
```diff
@@ -4,30 +4,51 @@
+    "build": "yarn run test:generate-output && build-storybook --quiet -c .storybook",
+    "lint": "eslint -c .eslintrc.js . && stylelint --config stylelint.config.js 'src/**/*.scss'",
     "test": "jest src",
+    "test:generate-output": "jest --json --outputFile=.jest-test-results.json",
```
What purpose is this serving?

seancdavis replied:
Nvm. I see it's used in the develop command. Do you ever run it individually?

phritolay replied:
I'm adding it to the `build` and `develop` scripts and removing it because it's not run individually.

seancdavis commented on package.json:
```diff
@@ -4,30 +4,51 @@
+    "@storybook/react": "next",
+    "classnames": "^2.2.6",
+    "css-loader": "^3.4.2",
+    "gatsby": "^2.18.12",
```
Make sure we have the most recent version of Gatsby. The version got downgraded — the caret means it'll likely be the same version, but wanting to check. Wondering if you copied and pasted from another project rather than installing and letting the newest versions write themselves?

phritolay replied:
Yes, this was something that probably occurred when I was trying to figure out the conflicts between the theme and some of our Gatsby projects.

seancdavis commented on src/components/layout/debug/media-queries/index.js:
```diff
@@ -0,0 +1,37 @@
+{
```
For now, let's put everything directly in the `src/components` directory. Until we have enough complexity to warrant further structure, I say we keep it simple.

seancdavis commented on src/components/seo/index.js:
```diff
@@ -27,16 +27,16 @@ const SEO = ({ description, og, imageUrl, title, titleTemplate, twitter, url })
-   * Meta description. Serves as a backup for og:desciption and
-   * twitter:desciption.
+   * Meta description. Serves as a backup for og:description and
+   * twitter:description.
```
Haha — thank you! I made this change upstream, too (after installing a spell checker). Woops!

seancdavis reviewed (changes requested):
A few comments. Let me know what you think.

seancdavis commented on src/components/link/notes.stories.mdx:
```diff
@@ -1,14 +1,10 @@
 import Link from "./"
 import { ExternalLink, InternalLink } from "./"
 
+<Meta title="Components/Link" component={Link} />
```
Looks like we lost the usage code here. [See here](https://deploy-preview-27--gatsby-theme-ample-components.netlify.com/?path=/story/components-link--page#usage).

seancdavis commented on src/components/layout/debug/media-queries/notes.stories.mdx:
```diff
@@ -0,0 +1,39 @@
+
+import DebugMediaQueries from "./"
+
+<Meta title="Utilities/Debug Media Queries" component={DebugMediaQueries} />
```
Here, too, feeling like we should just group everything together for now.

phritolay committed:
Merge branch 'master' into feature/media-queries-debugger-and-css-modules

phritolay committed:
move debug media queries to the components directory

add previews for the Link component
update Gatsby and other node modules
update develop and build scripts
condense exports on index.js

seancdavis committed:
Update index.stories.mdx
